fix(#490): child order of buy order was not sent to maker, but to taker

arkanoider opened the pull request:
@Catrya 

this is the fix for #490 .
Tested with a sell order and a buy order - now child order seems correctly sent to makers.

Catrya reviewed (commented):
Hi @arkanoider 
Now in purchase orders Mostrod sends the `new-order` message of the child order to the correct user, but when someone takes the child order that goes to `waiting-buyer-invoice` status, it sends the `add-invoice` message to the taker of the first order. In the database, in `creator_pubkey`, there is the trade key of the taker, not the maker (i mean the next trade key, no the key the used in 1st time)

In sales orders it still works well

arkanoider commented:
great...i will revert this. It's weird it did not work before, will look deeper.

Catrya reviewed (commented):
Hi @arkanoider mostro crashes when it receives the release action in a purchase range order.
```
2025-06-06T02:24:02.104833Z  INFO mostrod::util: Sending DM, Event ID: be5822df7adf89c9dca0dd339b88b66e097efeb9fb7854b4f8e1f1ff37825db4 with payload: "{\"order\":{\"version\":1,\"request_id\":null,\"trade_index\":null,\"id\":\"02f51039-bd0f-492b-989d-e0fd7695f806\",\"action\":\"released\",\"payload\":null}}"
thread 'main' panicked at src/app/release.rs:187:74:
called `Option::unwrap()` on a `None` value
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

arkanoider commented:
@Catrya seems working on my side. Tested also privacy orders and seems ok. I fixed and refactored the code for better readability. Improved privacy: now child order master seller/buyer key is salted in case of normal user, previously they were the same of parent order.
As usual you have the final sentence on tests! :wink:

Catrya reviewed (approved):
tACK great job @arkanoider
